### PR TITLE
Total count and configurable default per_page

### PR DESCRIPTION
**Pull request by jkorz**
## What is the current behavior?

Default per-page is 30 and set using a constant. Pagination meta does not include total number of records.

## What is the new behavior?

Implementers may now configure the default per-page by defining a method. Total number of records is included in the pagination meta.

## Checklist

Please make sure the following requirements are complete:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes /
  features)
- [ ] All automated checks pass (CI/CD)

### CODE REVIEW

**Comment by stas** [comment]
@jkorz there's already a PR related to the totals (stalled, for which I'm ashamed :see_no_evil:), please take a look at #29 

I like the idea of moving away from the constant, so I'd be happy to merge that part. But if I may suggest, I'd like us to keep the naming similar to what we have now, as in: `JSONAPI_PAGE_SIZE` to become `jsonapi_page_size`.

To move forward with this, I suggest we merge #29 and let you rebase your work to cover just the default page size setup.

How does that sound to you?

**Comment by jkorz** [comment]
@stas sounds good to me

**Comment by jkorz** [comment]
@stas I made the changes you requested.

File: lib/jsonapi/pagination.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def jsonapi_pagination(resources)
       original_params[:page] = original_params[:page].dup || {}
       original_url = request.base_url + request.path + '?'
 
-      pagination.each do |page_name, number|
+      pagination.delete_if{ |k, _v| k == :records }.each do |page_name, number|
```
Comment (stas) [inline lib/jsonapi/pagination.rb:45]: @jkorz , please consider instead `next if page_name == :records`, unless I'm missing something re what's going on here :see_no_evil:

**Comment by stas** [comment]
@jkorz this is good to go, let's just make sure the CI passes.

**Pull request merged**